example/CodeFuse/CommentCodePairsJava.gdl

```diff
@@ -0,0 +1,303 @@
+                }
+            }
+        }
+    }
+}
+}
+
+schema PublicVisitedElement extends Callable {
+    
+}
+
+//定义分母
+impl PublicVisitedElement {
+
+    @data_constraint
+    @inline
+    pub fn __all__(db: JavaDB) -> *PublicVisitedElement {
+        for (tmp in Callable(db)) {
+            yield PublicVisitedElement {
+                id : tmp.id
+            }
+        }
+    }
+    
+    //返回在上方的单行注释
+    pub fn getPossibleAboveComment(self) -> Comment {
+        for (c in Comment(__all_data__)) {
+            if (self.key_eq(c.getParent())) {
+                for (auto_tmp1 in self.getModifier()) {
+                    if (auto_tmp1.getLocation().getStartLineNumber() > c.getLocation().getStartLineNumber()) {
+                        return c
+                    }
+                }
+            }
+        }
+    }
+    
+    pub fn getDefinitionBody(self) -> string {
+        for (definitionBody in string::__undetermined_all__()) {
+            for (m in Method(__all_data__)) {
+                if (self.key_eq(m)) {
+                    if (definitionBody = m.getDefinitionBody()) {
+                        return definitionBody
+                    }
+                }
+            }
+            for (c in Constructor(__all_data__)) {
+                if (self.key_eq(c)) {
+                    if (definitionBody = c.getDefinitionBody()) {
+                        return definitionBody
+                    }
+                }
+            }
+        }
+    }
+}
+
+//找到注释与JavaDoc注释连在一起的情况，用于排除
+pub fn commentConnectDocComment(c: Comment) -> bool {
+    let (db = default_db()) {
+        for (i in int::__undetermined_all__(),
+            m in int::__undetermined_all__(),
+            l in int::__undetermined_all__()) {
+            for (j in JavadocComment(db)) {
+                if (getJavadocCommentLocInfo(j, i, m)) {
+                    if (l = m - 1) {
+                        if (getCommentLocInfo(c, i, l)) {
+                            return true
+                        }
+                    }
+                }
+            }
+        }
+    }
+}
+
+//找到注释所在文件和行号信息
+pub fn getCommentLocInfo(e: Comment, i: int, j: int) -> bool {
+    let (db = default_db()) {
+        if (i = e.getLocation().getFile().element_hash_id) {
+            if (j = e.getLocation().getStartLineNumber()) {
+                return true
+            }
+        }
+    }
+}
+
+//找到公共访问元素的所属文件和行号信息
+pub fn getPublicElementLocInfo(e: PublicVisitedElement, i: int, j: int) -> bool {
+    let (db = default_db()) {
+        if (i = e.getLocation().getFile().element_hash_id) {
+            if (j = e.getLocation().getStartLineNumber()) {
+                return true
+            }
+        }
+    }
+}
+
+//找到Javadoc注释所在文件和行号信息
+pub fn getJavadocCommentLocInfo(e: JavadocComment, i: int, j: int) -> bool {
+    let (db = default_db()) {
+        for (n in int::__undetermined_all__()) {
+                    if (i = e.getLocation().getFile().element_hash_id) {
```

Comment:
indentation in this line is not correct, make sure you are using 4 spaces instead of tab